week4/src/pages/MyPage.tsx

```diff
@@ -0,0 +1,110 @@
+import styled from "styled-components";
+import MypageContainer from "../components/Container"; 
+import Button from "../components/Button"; 
+import PasswordChangeForm from "../components/PasswordChangeForm"; 
+import { useEffect, useState } from "react";
+import { userInfo } from "../apis/userInfo"; 
+import { useParams } from "react-router-dom";
+import toggleIcon from "../assets/img/toggle.png"; 
+
+export interface UserType {
+  authenticationId: string;
+  nickname: string;
+  phone: string;
+}
+
+const UserProfile = () => {
+  const userId = useParams().id; 
+  const initialUserInfo = {
+    authenticationId: "",
+    nickname: "",
+    phone: "",
+  };
+  const [userDetails, setUserDetails] = useState<UserType>(initialUserInfo);
+
+  // 비밀번호 변경 토글
+  const [isPwdChangeVisible, setIsPwdChangeVisible] = useState(false);
+  const handleTogglePwdChange = () => {
+    setIsPwdChangeVisible((prev) => !prev);
+  };
+
+  // 사용자 정보 가져오기 
+  const fetchUserInfo = async () => {
+    if (userId) {
+      const info = await userInfo(userId);
+      setUserDetails(info);
+    }
+  };
+
+  useEffect(() => {
+    if (userDetails === initialUserInfo) fetchUserInfo();
+  }, [userDetails, initialUserInfo]);
```

Comment:
`userDetails`와 `initailUserInfo`가 같은 경우에만 사용자 정보를 가져오시는 이유가 궁금해요! 
만약 의도가 사용자 정보가 없는 상황일 때만 받아오려고 하신거라면, 사용자 정보가 없는 상황인 처음 마운트 될 때만 `fetchUserInfo`
함수를 조건문 없이 호출하면 어떨까 싶어요~
다른 이유가 있다면 궁금합니다!!

```suggestion
 useEffect(() => {
    fetchUserInfo();
  }, []);
```

Reply:
초기 사용자 정보(initialUserInfo)가 설정된 후, 실제 사용자 정보가 아직 불러와지지 않았을 경우에만 사용자 정보를 불러오려고 했었어요..! 대원님 의견 처럼 바꾸는게 좋을 거 같네요! 감사합니다!